### ISSUE TITLE
Paper: question about scoremean target

### ISSUE THREAD

**Issue by isseebx123** [lightvector/KataGo#153]
First of all, thank you for the experiments and English is not my first language so please excuse any mistakes

I looked at your paper and code and wondered what was used as the target data for scoremean. When I looked at the paper, I thought about using the expected value of final-score-distribution as the target data for scoremean.

![image](https://user-images.githubusercontent.com/26362491/76054951-a9295900-5fb5-11ea-8c08-9947c819e9ae.png)

But looking at model.py, scoremean seems to use the fourth value of globalTargetsNC.
```py
placeholders ["scoremean_target"] = features ["gtnc"] [:, 3]
```
And when I look at the cpp code that leaves the training data, the fourth value of globalTargetsNC seems to be made up of only accumulated scoreMean. The path I've tracked goes up the fillValueTDTargets function in reverse order and sees the use of node.stats.scoreMeanSum.
```cpp
static void fillValueTDTargets (const vector <ValueTargets> & whiteValueTargetsByTurn, int idx, Player nextPlayer, double nowFactor, float * buf) {
  // skip
  buf [3] = (float) score;
}
```
```cpp
bool Search::getNodeValues(const SearchNode& node, ReportedSearchValues& values) const {
  // skip
  double scoreMeanSum = node.stats.scoreMeanSum;
  // skip
  double scoreMean = scoreMeanSum / weightSum;
  // skip
  values.expectedScore = scoreMean;
  // skip
}
```
In my opinion, scoremean doesn't seem to be trained on the expected value of final-score-distribution.

Can you tell me if I am wrong or missing anything? Or can you explain how the scoremean target data is generated? Thank you.

**Comment by lightvector** [comment]
Yes, in the current tip of master, rather than training on the expected value of the final score distribution via self-prediction, it simply trains on the final score of the actual game (the same final score that the final score distribution is trained to predict). At different times during KataGo's development, the details of this have changed a little, but I don't think it matters very much exactly how you do it.

Changing to directly predict just the final score instead of self-predict the distribution was done around when Japanese rules and button Go support was added and it became a little difficult to think about how to make everything consistent with more complexity around even/odd and integer and non-integer scores depending on the rules. The score distribution target, although it still supplies a variance for the scoreMean, is now highly non-critical.  - if you wanted you could get rid of the distribution entirely, and scoreMean would still learn perfectly fine since now it's learning directly from the game result instead of learning by regularizing towards something that's learning from the game result.

Due to the Huber loss it's not *quite* the same as the mean, but generally it will be very very close.

**Comment by isseebx123** [comment]
I was surprised that the score mean and distribution are not highly important and are well trained without some.

Thank you for the quick and friendly explanation. I understood it perfectly.
